### PR TITLE
refactor: move `DRACOExporter` into TypeScript

### PR DESCRIPTION
**Pull request by shuta13**
### What

Part of #12 

This is a PR of `DRACOExporter` migration to TypeScript.

### Checklist

- [ ] Ready to be merged

### CODE REVIEW

File: src/exporters/DRACOExporter.ts [hunk 1]
```diff
@@ -12,50 +14,63 @@
...
-/* global DracoEncoderModule */
-
-const DRACOExporter = () => {}
-
-DRACOExporter.prototype = {
-  constructor: DRACOExporter,
-
-  parse: function (object, options) {
-    if (object.isBufferGeometry === true) {
+class DRACOExporter {
```
Comment (shuta13) [inline src/exporters/DRACOExporter.ts:17]: ref: https://github.com/mrdoob/three.js/blob/069e6c69490f8be5a31c9c480bd269d667fda9d8/examples/jsm/exporters/DRACOExporter.js

File: src/exporters/DRACOExporter.ts [hunk 2]
```diff
@@ -12,50 +14,63 @@
  *  - exportNormals
  */
 
-/* global DracoEncoderModule */
```
Comment (shuta13) [inline src/exporters/DRACOExporter.ts:15]: In the original version of `three.js`, undefined `DracoEncoderModules` were notified to Linter and others by writing `/* global DracoEncoderModules */`.

However, when I migrated `DRACOExporter` to TypeScript, I found that I could not use `/* global DracoEncoderModules */`, so I changed it to not check the type by adding `@ts-ignore`. Is this the correct way to handle this?

**Review by shuta13 — COMMENTED** [review]
I moved `DRACOExporter` into TypeScript and made some modifications.

File: src/custom.d.ts [hunk 3]
```diff
@@ -3,3 +3,91 @@ declare module 'mmd-parser' {
...
+}
+
+class DracoInt8Array {
+  public GetValue: (index: number) => number
+}
+interface IDracoInt8Array {
+  new (): DracoInt8Array
+}
+
+type DracoEncoderModule = () => {
```
Comment (shuta13) [inline src/custom.d.ts:83]: ref: https://github.com/google/draco/blob/8a979f79a5f139880f17f296ace90bcfff025c4b/src/draco/javascript/emscripten/draco_web_encoder.idl

**Comment by shuta13** [comment]
Updated to add type definition for DracoEncoderModule. Please check it as well. : https://github.com/pmndrs/three-stdlib/pull/76/commits/fa96b430747bcca9ba5c26b48dfef4413ef4480e

**Comment by donmccurdy** [comment]
Just as FYI: https://github.com/google/draco/issues/689 

Would shared types like `@types/draco3dgltf` be helpful to this project as well, if they were published?

**Comment by shuta13** [comment]
@donmccurdy 
Thanks for the information!

Currently, we are working on adding type definitions of draco to `custom.d.ts`, but it would be nice if DefinitelyTyped shared the types. However, I'm not a maintainer of `three-stdlib`, so I'll follow the maintainer's decision.

@joshuaellis 
What do you think about donmccurdy's suggestion?

**Comment by joshuaellis** [comment]
> Just as FYI: [google/draco#689](https://github.com/google/draco/issues/689)
> 
> Would shared types like `@types/draco3dgltf` be helpful to this project as well, if they were published?

@donmccurdy that would be incredibly useful, that way we don't have to figure it out and update!!

**Comment by donmccurdy** [comment]
Opened PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54995

**Comment by donmccurdy** [comment]
^Might be helpful if someone could review this PR and/or type-check it against your code, the DefinitelyTyped bot seems to be encouraging external reviews. https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54995#issuecomment-894594701 :)

**Comment by joshuaellis** [comment]
Cheers @donmccurdy ill take a look today 😄

**Comment by joshuaellis** [comment]
Works great @donmccurdy, I have left a couple of comments on your PR 😄 thanks again for sorting this out, makes a great addition to the ecosystem!

**Comment by joshuaellis** [comment]
@shuta13 we should update this pr to include the type package.

File: src/exporters/DRACOExporter.ts [hunk 4]
```diff
@@ -47,15 +63,16 @@ DRACOExporter.prototype = {
...
     }
 
-    if (object.isMesh === true) {
+    if (object instanceof Mesh && object.isMesh) {
       builder = new dracoEncoder.MeshBuilder()
       dracoObject = new dracoEncoder.Mesh()
 
-      var vertices = geometry.getAttribute('position')
+      const vertices = geometry.getAttribute('position')
+      /// @ts-expect-error
```
Comment (shuta13) [inline src/exporters/DRACOExporter.ts:75]: Functions and methods that do not exist in the type are marked with `@ts-expect-error`, although there are several others

**Review by shuta13 — COMMENTED** [review]
@joshuaellis 

Sorry for the delay in response. Please check.

File: src/exporters/DRACOExporter.ts [hunk 5]
```diff
@@ -47,15 +63,16 @@ DRACOExporter.prototype = {
...
     }
 
-    if (object.isMesh === true) {
+    if (object instanceof Mesh && object.isMesh) {
       builder = new dracoEncoder.MeshBuilder()
       dracoObject = new dracoEncoder.Mesh()
 
-      var vertices = geometry.getAttribute('position')
+      const vertices = geometry.getAttribute('position')
+      // @ts-expect-error
```
Comment (shuta13) [inline src/exporters/DRACOExporter.ts:75]: There are several others, but type errors are marked with `@ts-expect-error`

**Pull request merged**